pkg/controller/hibernation/hibernation_controller_test.go

```diff
@@ -406,11 +678,15 @@ func TestReconcile(t *testing.T) {
 				testcd.WithAnnotation(constants.HiveFakeClusterAnnotation, "true")),
 			cs: csBuilder.Build(),
 			validate: func(t *testing.T, cd *hivev1.ClusterDeployment) {
-				cond := getHibernatingCondition(cd)
+				cond, runCond := getHibernatingAndRunningConditions(cd)
 				require.NotNil(t, cond)
 				assert.Equal(t, hivev1.HibernatingHibernationReason, cond.Reason)
 				assert.Equal(t, corev1.ConditionTrue, cond.Status)
 				assert.Equal(t, "Fake cluster is stopped", cond.Message)
+				require.NotNil(t, runCond)
+				assert.Equal(t, corev1.ConditionFalse, runCond.Status)
+				assert.Equal(t, hivev1.StoppingOrHibernatingReadyReason, runCond.Reason)
+				assert.Equal(t, hivev1.HibernatingHibernationReason, cd.Status.PowerState)
```

Comment:
I didn't look at *all* the tests, but it looks like many of these assertions could be factored out into fields of the test config, and validated by the main driver. These tests are supposed to be declarative as much as possible; including a validation function is kinda cheating.

(Note: I personally don't love the declarative style, especially with so many test cases, because it's really hard to put a breakpoint on a specific test -- I always have to comment out all the preceding cases, which sucks.)